### PR TITLE
book: Add chapter about troubleshooting

### LINKED ISSUE #331: thread 'main' panicked at 'No resource with the given id'
I have the following code:
```rust
    fn update(&mut self, _ctx: &mut Context, _dt: Duration) -> GameResult<()> {
        self.update(InputEvent::None);
        self.dispatcher.dispatch(&mut self.world.res);

        Ok(())
    }
```
dispatcher is a `Dispatcher<'static, 'static>` and world is a `World`.

Here's the StackTrace:
thread 'main' panicked at 'No resource with the given id', src\libcore\option.rs:874:4
stack backtrace:
```
   0:     0x7ff6e9dbfa8d - std::sys::windows::backtrace::unwind_backtrace
                               at C:\projects\rust\src\libstd\sys\windows\backtrace\mod.rs:65
   1:     0x7ff6e9db8077 - std::sys_common::backtrace::_print
                               at C:\projects\rust\src\libstd\sys_common\backtrace.rs:68
   2:     0x7ff6e9db5da3 - std::panicking::default_hook::{{closure}}
                               at C:\projects\rust\src\libstd\panicking.rs:383
   3:     0x7ff6e9db58ec - std::panicking::default_hook
                               at C:\projects\rust\src\libstd\panicking.rs:397
   4:     0x7ff6e9db6385 - std::panicking::rust_panic_with_hook
                               at C:\projects\rust\src\libstd\panicking.rs:577
   5:     0x7ff6e9db6190 - std::panicking::begin_panic<alloc::string::String>
                               at C:\projects\rust\src\libstd\panicking.rs:538
   6:     0x7ff6e9db60e1 - std::panicking::begin_panic_fmt
                               at C:\projects\rust\src\libstd\panicking.rs:522
   7:     0x7ff6e9db6059 - std::panicking::rust_begin_panic
                               at C:\projects\rust\src\libstd\panicking.rs:498
   8:     0x7ff6e9dcca1c - core::panicking::panic_fmt
                               at C:\projects\rust\src\libcore\panicking.rs:71
   9:     0x7ff6e9dccdca - core::option::expect_failed
                               at C:\projects\rust\src\libcore\option.rs:874
  10:     0x7ff6e9c5794b - shred::system::{{impl}}::run_now<plan_col::game::entities::MoveToDestination>
                               at C:\Users\itmuckel\.cargo\registry\src\github.com-1ecc6299db9ec823\shred-0.5.1\src\system.rs:25
  11:     0x7ff6e9c6b36c - shred::dispatch::dispatcher::Dispatcher::dispatch
                               at C:\Users\itmuckel\.cargo\registry\src\github.com-1ecc6299db9ec823\shred-0.5.1\src\dispatch\dispatcher.rs:33
  12:     0x7ff6e9c6155e - plan_col::run
                               at C:\Users\itmuckel\Documents\TEMP\Rust\plan_col\src\lib.rs:15
  13:     0x7ff6e9dc92a1 - panic_unwind::__rust_maybe_catch_panic
                               at C:\projects\rust\src\libpanic_unwind\lib.rs:101
  14:     0x7ff6e9db0fbb - std::rt::lang_start
                               at C:\projects\rust\src\libstd\rt.rs:50
  15:     0x7ff6e9ddc413 - __scrt_common_main_seh
                               at f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:283
  16:     0x7ffc667b1fe3 - BaseThreadInitThunk
```

Without the call to dispatch the program runs, but is of course useless. 😆 

In the docs it says that `dispatch` panics, when resources are borrowed, but I don't think I have done something like that. What resources is this about?

Any ideas?

### PR DESCRIPTION
**Pull request by Rhuagh**
Fixes #331. Will need rebasing after the other book PRs are merged.

---
This change is [Reviewable](https://reviewable.io/reviews/slide-rs/specs/399)

### CODE REVIEW

File: book/src/12_tracked.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+```
+
+There are three different event types that we can subscribe to:
+
+* InsertedFlag - will be sent whan a component is added to the storage
+* ModifiedFlag - will be sent when a component is fetched mutable from the storage
+* RemovedFlag - will be sent when a component is removed from the storage
+
+Note that because of how `ModifiedFlag` works, if you iterate mutable over a
```
Comment (torkleyy) [inline book/src/12_tracked.md]: I think you need to use an adverb here (mutably)

File: book/src/12_tracked.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+There are three different event types that we can subscribe to:
+
+* InsertedFlag - will be sent whan a component is added to the storage
+* ModifiedFlag - will be sent when a component is fetched mutable from the storage
+* RemovedFlag - will be sent when a component is removed from the storage
+
+Note that because of how `ModifiedFlag` works, if you iterate mutable over a 
+component storage using `Join`, all entities that are fetched by the `Join` will
+be flagged as modified regardless of if anything was updated in them.
```
Comment (torkleyy) [inline book/src/12_tracked.md]: "even if nothing was updated" is a bit more natural

File: book/src/13_troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Troubleshooting
+
+## `no resource with the given id`
```
Comment (torkleyy) [inline book/src/13_troubleshooting.md]: I'm not sure the panic message still contains these words

File: book/src/13_troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+This panic will occur whenever a `System` is first dispatched, and one or 
+more of the components and/or resources it uses is missing from `World`.
+
+There are a few main reasons for this occurring:
+
+* Forgetting to call `setup` after building a `Dispatcher` or `ParSeq`. Make 
+  sure this is always run before the first dispatch.
+* Not adding mandatory resources to `World`. You can usually find these by 
+  searching for occurrences of `ReadExpect` and `WriteExpect`.
+* Manually requesting components/resources from `World` (not inside a `System`),
```
Comment (torkleyy) [inline book/src/13_troubleshooting.md]: Most notably with the `EntityBuilder`

**Comment by Rhuagh** [comment]
Will push all these in a single PR.

**Pull request closed without merging**